common/metrics/grpc.go

```diff
@@ -35,7 +36,14 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-type baggageContextKey struct{}
+type (
+	baggageContextKey struct{}
+
+	// contextBaggage is used to propagate metrics across single gRPC call within server
+	contextBaggage struct {
+		CountersInt *sync.Map
```

Comment:
some blogs says sync.Map is slow, can you do a bench?
my guess, sync.Map can be performant if there are tons of concurrent read and write, which is not this use case